NFD master during filterExtendedResources when feature not available

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

```sh
E0421 12:52:31.813082       1 nfd-master.go:912] failed to process Rule "rdt l3 closid extended rule": feature "cpu.rdt" not available
panic: runtime error: index out of range [1] with length 1

goroutine 131 [running]:
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).filterExtendedResources(0xc000557578?, 0xc0006aa020, 0xc00088edb0?)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:709 +0xc57
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).refreshNodeFeatures(0xc0005a5900, 0xc0005577f0?, {0xc000632b40, 0x2b}, 0x2?, 0x194bc40?, 0xc0006aa020?)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:761 +0x2c5
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).nfdAPIUpdateOneNode(0xc0005a5900, {0xc000632b40, 0x2b})
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:675 +0x487
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).nfdAPIUpdateAllNodes(0xc0005a5900)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:607 +0x170
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).nfdAPIUpdateHandler(0xc0005a5900)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:335 +0x1cc
created by sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).Run
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:225 +0x6fe
```

**What you expected to happen**:

Log error, without crashing app

**How to reproduce it (as minimally and precisely as possible)**:

Create such rule:
```yaml
apiVersion: nfd.k8s-sigs.io/v1alpha1
kind: NodeFeatureRule
metadata:
  name: rdt-l3-closid-extended-rule
spec:
  rules:
    - name: "rdt l3 closid extended rule"
      extendedResources:
        feature.node.kubernetes.io/rdtl3_closid: "@RDTL3CA_NUM_CLOSID"
      matchFeatures:
        - feature: cpu.rdt
          matchExpressions:
            RDTL3CA_NUM_CLOSID: {op: Exists}
```

**Anything else we need to know?**:

Running from `v0.13.0` tag not master branch.

--- a/pkg/nfd-master/nfd-master.go
+++ b/pkg/nfd-master/nfd-master.go
@@ -713,6 +713,10 @@ func (m *nfdMaster) filterExtendedResources(features *nfdv1alpha1.Features, exte
 		if strings.HasPrefix(capacity, "@") {
 			// capacity is a string in the form of attribute.featureset.elements
 			split := strings.SplitN(capacity[1:], ".", 3)
+			if len(split) != 3 {
+				klog.Errorf("capacity %s is not in the form of '@domain.feature.element',. Ignoring Extended Resource %q", capacity, extendedResource)
+				continue
+			}
 			featureName := split[0] + "." + split[1]
 			elementName := split[2]
 			attrFeatureSet, ok := features.Attributes[featureName]

